src/redux/slices/QuranReader/translations.ts

```diff
@@ -1,14 +1,55 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
-const DEFAULT_TRANSLATIONS = [20, 131];
+export const DEFAULT_TRANSLATIONS = [20, 131];
 
-const initialState: number[] = DEFAULT_TRANSLATIONS;
+export type TranslationsSettings = {
+  selectedTranslations: number[];
+  isUsingDefaultTranslations: boolean;
+};
+
+/**
+ * This is to detect whether the user has changed the default translations or not on each translation change.
+ * The user will have changed the default translations in 2 cases:
+ *  1. the length of the array of IDs of the translations the user has selected differs from the length of the array of default IDs.
+ *  2. the length of both arrays are the same but one ID exists in one but not in the other.
+ *
+ * [NOTE]: We could have used the easier JSON.stringify(DEFAULT_TRANSLATIONS) === JSON.stringify(newTranslations)
+ *         comparison but there is a corner case when the user changes the default translations then
+ *         re-selects them which results in the same array as the default one but reversed
+ *         e.g. instead of [20, 131] it becomes [131, 20] and the JSON comparison will fail
+ *         although the BE API's response will have the translations in the same order in both cases
+ *         which is by ID ascendingly (e.g. 20 then 131 even if we set the translations' param as "131, 20").
+ * @param {number[]} newTranslations
+ * @returns {boolean}
+ */
+const isUsingDefaultTranslations = (newTranslations: number[]): boolean => {
+  // if the lengths are different, it means the user has changed the default translations.
+  if (newTranslations.length !== DEFAULT_TRANSLATIONS.length) {
+    return false;
+  }
+  for (let i = 0; i < newTranslations.length; i += 1) {
+    // if the current translation ID is not inside the default IDs' array, it means the user has changed the default translations.
+    if (DEFAULT_TRANSLATIONS.indexOf(newTranslations[i]) === -1) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
I like this logic, 2 things: 

1. It's simpler to just do `_.isEqual(DEFAULT_TRANSLATIONS.sort(), newTranslations.sort());` in 1 line of code :) 

2. In general, try to move the helper functions to the bottom of the file (before the styles), to make the "main" part easier to follow

Reply:
Oh I forgot about the mighty swiss knife `lodash` :D  thanks for the suggestion.